client/hub/client.go

```
@@ -162,12 +164,22 @@ func (c *hubClient) getTLSConfig() *tls.Config {
 	// If CACertData is present use it
 	if certData.CACertData != "" {
 		var pool *x509.CertPool
-		var err error
+
+		decodedCACertData, err := base64.StdEncoding.DecodeString(certData.CACertData)
+		if err != nil {
+			log.Infof("unable to use custom cert when taking to '%s' endpoint. Error: %s", c.tanzuHubEndpoint, err.Error())
+			return nil
```

should it be `return err`?

---

In this case, I was thinking about returning `nil` for `*tls.Config` with the log message, so interacting with the client will throw error if the cert verification fails instead of returning error here.

---

Ok. Got. Thank you!